### PR TITLE
Feature/validate required fields

### PR DESCRIPTION
**Pull request by allan-railz**

### CODE REVIEW

**Comment by allan-railz** [comment]
Remove accountingMethod until this is publicly available.

File: packages/components/src/elements/bank-accounts/bank-accounts.utils.ts [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
 export const getReportData = async ({
   filter,
-}: RVReportRequestDateParameter): Promise<RVFormattedBankAccountsResponse> => {
+}: RVReportRequestParameter): Promise<RVFormattedBankAccountsResponse> => {
   let reportData;
   try {
     reportData = await RequestServiceInstance.getReportData({
-      reportType: filter.reportType,
-      filter: pick(filter, [RVParams.SERVICE_NAME, RVParams.BUSINESS_NAME]),
+      filter: pick(filter, [RVParams.SERVICE_NAME, RVParams.BUSINESS_NAME, RVParams.REPORT_TYPE]),
```
Comment (abisolaadeniran) [inline packages/components/src/elements/bank-accounts/bank-accounts.utils.ts:19]: bank account does not need report type

**Review by abisolaadeniran — CHANGES_REQUESTED** [review]
Apart from the other comments below:
1. when wrong parameters are passed, we are supposed to show no data
its not showing anything. Passed businessName and serviceName
Screen Shot 2022-05-18 at 9 07 05 AM

2. When wrong serviceName is passed, we make an API call, we should check if its a valid serviceName Screen Shot 2022-05-18 at 9 09 29 AM

**Comment by railz-dev** [comment]
All handled.

File: packages/components/src/elements/bank-accounts/bank-accounts.utils.ts (hunk 1 above)
Comment (allanhal) [reply packages/components/src/elements/bank-accounts/bank-accounts.utils.ts:19]: Do a URLMapping on the reports/requests

Comment (allanhal) [reply packages/components/src/elements/bank-accounts/bank-accounts.utils.ts:20]: Done

**Comment by abisolaadeniran** [comment]
1. we are still not validating if serviceName is a valid service name and showing correct info, for Bills, Invoices and Railz Score, we are showing Oops, something went wrong instead of No data available
Screen Shot 2022-05-20 at 11 52 00 AM

2. If no filter is passed, we are not logging any info letting them know we did not get the required data, also we are not showing the No data screen. Per requirement initially, the only time we won't load is if we don't get access token and we should definitely log if not passed
3. We are defaulting to plaid for bankAccounts even if they passed the wrong serviceName
4.  We are still sending reportType to bankAccounts API

**Comment by railz-dev** [comment]
> Apart from the other comments below:
> 
> 1. when wrong parameters are passed, we are supposed to show no data
>    its not showing anything. Passed businessName and serviceName
> 
> Screen Shot 2022-05-18 at 9 07 05 AM
> 
> 2. When wrong serviceName is passed, we make an API call, we should check if its a valid serviceName Screen Shot 2022-05-18 at 9 09 29 AM

Addressed.

**Comment by railz-dev** [comment]
> 1. we are still not validating if serviceName is a valid service name and showing correct info, for Bills, Invoices and Railz Score, we are showing Oops, something went wrong instead of No data available
>    Screen Shot 2022-05-20 at 11 52 00 AM
> 2. If no filter is passed, we are not logging any info letting them know we did not get the required data, also we are not showing the No data screen. Per requirement initially, the only time we won't load is if we don't get access token and we should definitely log if not passed
> 3. We are defaulting to plaid for bankAccounts even if they passed the wrong serviceName
> 4. We are still sending reportType to bankAccounts API

Addressed.

**Pull request closed without merging**